.github/workflows/r.yml

```diff
@@ -329,6 +329,24 @@ jobs:
         if: ${{ matrix.config.rtools == 35 }}
         shell: Rscript {0}
         run: install.packages("cpp11", type = "source")
+      - name: Prune dependencies (on R 3.6)
+        if: ${{ matrix.config.rtools == 35 }}
+        shell: Rscript {0}
+        run: |
+            # To prevent the build from timing out, let's prune some optional deps (and their possible version requirements)
+            setwd("r")
+            d <- read.dcf("DESCRIPTION")
+            to_prune <- c("duckdb", "DBI", "dbplyr", "decor", "knitr", "rmarkdown", "pkgload", "reticulate")
+            pattern <- paste0("\\n?", to_prune, " (\\(.*\\))?,?", collapse = "|")
+            d[,"Suggests"] <- gsub(pattern, "", d[,"Suggests"])
+            write.dcf(d, "DESCRIPTION")
+      - name: Restore R package cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ env.R_LIBS_USER }}/*
+          key: ${{ steps.get-version.outputs.os-version }}-${{ steps.get-version.outputs.r-version }}-${{inputs.cache-version }}-${{ hashFiles('r/DESCRIPTION') }}
+          restore-keys: ${{ steps.get-version.outputs.os-version }}-${{ steps.get-version.outputs.r-version }}-${{inputs.cache-version }}-
       - name: Install R package dependencies
         shell: Rscript {0}
         run: |
```

Comment:
Not sure if this will help what you are aiming for here, but [processx now has binaries for R-devel on Windows](https://cran.r-project.org/web/packages/processx/index.html), so we can uncomment this:

https://github.com/apache/arrow/blob/64f341b8a9b75a306ace4fd897dd0c56e094f772/.github/workflows/r.yml#L353

Reply:
Thanks for the suggestion, though I don't think that'll help here since this is building 3.6 and CRAN only builds binaries for release (4.1, soon to be 4.2) + old release (4.0)